### PR TITLE
DOC: fix docstring for pandas.read_parquet

### PR DESCRIPTION
**Pull request by igblackadder**
- [ ] closes #xxxx
- [ ] tests added / passed
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

```
################################################################################
####################### Docstring (pandas.read_parquet)  #######################
################################################################################

Load a parquet object from the file path, returning a DataFrame.

.. versionadded 0.21.0

Parameters
----------
path : str
    File path.
engine : {'auto', 'pyarrow', 'fastparquet'}, default 'auto'
    Parquet library to use. If 'auto', then the option
    ``io.parquet.engine`` is used. The default ``io.parquet.engine``
    behavior is to try 'pyarrow', falling back to 'fastparquet' if
    'pyarrow' is unavailable.
columns : list, default=None
    If not None, only these columns will be read from the file.

    .. versionadded 0.21.1
**kwargs
    Any additional kwargs are passed to the engine.

Returns
-------
DataFrame
    A parquet file is returned as two-dimensional data structure with
    labeled axes.

See Also
--------
to_parquet : Write DataFrame to a parquet file.
read_csv : Read a comma-separated values (csv) file into DataFrame.

Examples
--------
>>> pd.read_parquet('data.parquet')  # doctest: +SKIP

################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.read_parquet" correct. :)
```

### CODE REVIEW

File: pandas/io/parquet.py [hunk 1]
```diff
@@ -277,6 +277,17 @@ def read_parquet(path, engine='auto', columns=None, **kwargs):
     Returns
     -------
     DataFrame
+        A parquet file is returned as two-dimensional data structure with
+        labeled axes.
+
+    See Also
+    --------
+    to_parquet : Write DataFrame to a parquet file.
```
Comment (TomAugspurger) [inline pandas/io/parquet.py]: Does this get linked correctly by numpydoc / sphinx? Or does it need to be `DataFrame.to_parquet`?

File: pandas/io/parquet.py [hunk 2]
```diff
@@ -277,6 +277,17 @@ def read_parquet(path, engine='auto', columns=None, **kwargs):
     Returns
     -------
     DataFrame
+        A parquet file is returned as two-dimensional data structure with
```
Comment (TomAugspurger) [inline pandas/io/parquet.py]: (To the other maintainers) do we have a general policy on these? I don't think this description adds much beyond just having the return type.

**Review by TomAugspurger — COMMENTED** [review]
Could you add a

File: pandas/io/parquet.py (hunk 2 above)
Comment (WillAyd) [reply pandas/io/parquet.py]: We do technically require a description for the Returns but I would be on favor of relaxing hat requirement. Agreed this doesn’t add value and may not in most instances

File: pandas/io/parquet.py (hunk 1 above)
Comment (igblackadder) [reply pandas/io/parquet.py]: It appears it does need the `DataFrame.` prefix to link correctly. A brief bit of searching shows this problem exists on at least two other pages (both with `to_csv` in the See Also sections of `read_csv` and `read_fwf`). Is it better to commit a change for all three as part of this PR or just fix this one and create a new PR for the other two?

Comment (WillAyd) [reply pandas/io/parquet.py]: Just fix this for now and leave the other two. Not sure if your check was only on this module but would be surprised if we only had two across the codebase, so would be fine to have a comprehensive PR separate to address

File: pandas/io/parquet.py (hunk 2 above)
Comment (WillAyd) [reply pandas/io/parquet.py]:
```suggestion
        DataFrame with appropriate data.
```

As mentioned probably shouldn't need this at all, but can at least shorten to get CI to pass in the meantime

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/io/parquet.py (hunk 1 above)
Comment (igblackadder) [reply pandas/io/parquet.py]:
```suggestion
    DataFrame.to_parquet : Write DataFrame to a parquet file.
```

File: pandas/io/parquet.py [hunk 3]
```diff
@@ -277,6 +277,17 @@ def read_parquet(path, engine='auto', columns=None, **kwargs):
     Returns
     -------
     DataFrame
+        DataFrame with appropriate data.
+        labeled axes.
```
Comment (TomAugspurger) [inline pandas/io/parquet.py]: Is this line deliberate?

Comment (igblackadder) [reply pandas/io/parquet.py]:
```suggestion
```

Comment (igblackadder) [reply pandas/io/parquet.py]: My bad, thank you for catching that

**Review by WillAyd — APPROVED** [review]

File: pandas/io/parquet.py [hunk 4]
```diff
@@ -260,8 +260,8 @@ def read_parquet(path, engine='auto', columns=None, **kwargs):
 
     Parameters
     ----------
-    path : string
-        File path
+    path : str
```
Comment (TomAugspurger) [inline pandas/io/parquet.py]: Should this be path-like (anything implementing __fspath__)? I suppose that's up to the engine. I don't think it's correct to say that only strings are supported here.

Comment (jreback) [reply pandas/io/parquet.py]: I *believe* pyarrow accepts pathlike (cc @wesm)  and fastparquet (cc @martindurant)

Comment (wesm) [reply pandas/io/parquet.py]: yes

**Comment by WillAyd** [comment]
Closing in favor of #25164 which appears more active

**Pull request closed without merging**